sdk/eventhubs/azure_messaging_eventhubs_checkpointstore_blob/src/checkpoint_store.rs

```diff
@@ -0,0 +1,324 @@
+        BlockBlobClientUploadResultHeaders, ListBlobsIncludeItem,
+    },
+    BlobContainerClient,
+};
+use futures::TryStreamExt;
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+use time::OffsetDateTime;
+use tracing::{debug, warn};
+
+/// Azure Blob Storage implementation of the [`CheckpointStore`] trait.
+///
+/// This implementation stores checkpoint and ownership information in Azure Blob Storage,
+/// providing durable persistence for Event Hub event processing state.
+#[derive(Clone)]
+pub struct BlobCheckpointStore {
+    blob_container_client: Arc<BlobContainerClient>,
+}
+
+impl Debug for BlobCheckpointStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BlobCheckpointStore").finish()
+    }
+}
+
+const OWNER_ID: &str = "ownerid";
+const OFFSET: &str = "offset";
+const SEQUENCE_NUMBER: &str = "sequencenumber";
+
+impl BlobCheckpointStore {
+    /// Creates a new blob checkpoint store.
+    ///
+    /// # Arguments
+    ///
+    /// * `blob_service_client` - The Azure Blob Storage service client
+    pub fn new(blob_container_client: BlobContainerClient) -> Arc<Self> {
+        Arc::new(Self {
+            blob_container_client: Arc::new(blob_container_client),
+        })
+    }
+
+    fn process_storage_response_metadata(
+        last_modified: Option<String>,
+        etag: Option<String>,
+    ) -> Result<(Option<OffsetDateTime>, Option<Etag>)> {
+        let lm = match last_modified {
+            Some(lm) => Some(parse_rfc7231(lm.as_str())?),
+            None => None,
+        };
+        Ok((lm, etag.map(Etag::from)))
+    }
+
+    async fn set_metadata_on_blob(
+        &self,
+        blob_name: &str,
+        metadata: Option<HashMap<String, String>>,
+        etag: Option<Etag>,
+    ) -> Result<(Option<OffsetDateTime>, Option<Etag>)> {
+        let blob_client = self
+            .blob_container_client
+            .blob_client(blob_name.to_string());
+        let mut options = BlobClientSetMetadataOptions::default();
+        if let Some(metadata) = &metadata {
+            options.metadata = Some(metadata.clone());
+        }
+        options.if_match = etag.clone().map(|e| e.to_string());
+        let result = blob_client.set_metadata(Some(options)).await;
+        match result {
+            Ok(r) => Ok(Self::process_storage_response_metadata(
+                r.headers().get_optional_string(&LAST_MODIFIED),
+                r.headers().get_optional_string(&ETAG),
+            )?),
+            Err(e) => {
+                match e.http_status() {
+                    Some(StatusCode::PreconditionFailed) => {
+                        // If the blob has been modified since we last read it, return a NotFound error
+                        Err(e)
+                    }
+                    Some(StatusCode::NotFound) => {
+                        warn!("Blob {blob_name} not found, creating.");
```

Comment:
This doesn't feel like a warning, which would often show by default. Maybe info or even debug. It's part of the intent and normal course of this implementation, right? If it doesn't exist, create it.